pkg/status/singletonstatus.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2024 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	"github.com/kubestellar/kubestellar/pkg/util"
+)
+
+// Should be wr-locked
+type singletonState struct {
+	wObjSync map[util.ObjectIdentifier]bool
+}
+
+func (ss *singletonState) Get(id util.ObjectIdentifier) bool {
+	return ss.wObjSync[id]
+}
+
+func (ss *singletonState) Set(id util.ObjectIdentifier) {
+	ss.wObjSync[id] = true
+}
+
+func (ss *singletonState) Unset(id util.ObjectIdentifier) {
+	ss.wObjSync[id] = false
```

Comment:
What deletes entries from this map?
If an entry is never deleted then this is a memory leak.